lib/packExternalModules.js

```diff
@@ -182,36 +182,42 @@ function isExternalModule(module) {
   return _.startsWith(module.identifier(), 'external ') && !isBuiltinModule(getExternalModuleName(module));
 }
 
+/**
+ * Gets the module issuer. The ModuleGraph api does not exists in webpack@4
+ * so falls back to using module.issuer.
+ */
+function getIssuerCompat(moduleGraph, module) {
+  if (moduleGraph != null) {
+    return moduleGraph.getIssuer(module);
+  } else {
+    return module.issuer;
+  }
+}
+
 /**
  * Find the original module that required the transient dependency. Returns
  * undefined if the module is a first level dependency.
+ * @param {Object} moduleGraph - Webpack module graph
  * @param {Object} issuer - Module issuer
  */
-function findExternalOrigin(issuer) {
+function findExternalOrigin(moduleGraph, issuer) {
   if (!_.isNil(issuer) && _.startsWith(issuer.rawRequest, './')) {
-    return findExternalOrigin(issuer.issuer);
+    return findExternalOrigin(moduleGraph, getIssuerCompat(moduleGraph, issuer));
   }
   return issuer;
 }
 
-function getExternalModules(stats) {
-  if (!stats.compilation.chunks) {
-    return [];
-  }
+function getExternalModules({ compilation }) {
   const externals = new Set();
-  for (const chunk of stats.compilation.chunks) {
-    if (!chunk.modulesIterable) {
-      continue;
-    }
-
-    // Explore each module within the chunk (built inputs):
-    for (const module of chunk.modulesIterable) {
-      if (isExternalModule(module)) {
-        externals.add({
-          origin: _.get(findExternalOrigin(module.issuer), 'rawRequest'),
-          external: getExternalModuleName(module)
-        });
-      }
+  for (const module of compilation.modules) {
```

Comment:
Just wondering if this is compatible with webpack 4? (or if it's related to what you said about not testing it on webpack 4 already?)

Reply:
It should be https://github.com/webpack/webpack/blob/webpack-4/lib/Compilation.js#L494. But yeah ideally someone would test on webpack 4 just to make sure.

Reply:
I just tested and it worked fine on webpack4